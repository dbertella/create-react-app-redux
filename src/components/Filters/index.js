import React from 'react'
import { Field, reduxForm } from 'redux-form'

const Filters = (props) => {
  const { handleSubmit, reset, change } = props
  const onChange = (e) => {
    const p = new Promise((resolve) => resolve(change('filterArtist', e.target.name, e.target.value)));
    p.then(() => handleSubmit());
  };
  return (
    <form onSubmit={handleSubmit}>
      Filters:
      <ul>
        <li>
          age range
          <Field component="input" type="text" placeholder="min" name="ageMin" onBlur={handleSubmit} />
          <Field component="input" type="text" placeholder="min" name="ageMax" onBlur={handleSubmit} />
        </li>
        <li>
          rate range
          <Field component="input" type="text" placeholder="min" name="rateMin" onBlur={handleSubmit} />
          <Field component="input" type="text" placeholder="min" name="rateMax" onBlur={handleSubmit} />
        </li>
        <li>
          gender:
          <label><Field component="input" type="radio" name="gender" value="" onClick={onChange} />none</label>
          <label><Field component="input" type="radio" name="gender" value="M" onClick={onChange} />M</label>
          <label><Field component="input" type="radio" name="gender" value="F" onClick={onChange} />F</label>
        </li>
      </ul>
      <button type="submit" onClick={reset}>Clear Filters</button>
    </form>
  )
}

export default reduxForm({
  form: 'filterArtist'
})(Filters)
