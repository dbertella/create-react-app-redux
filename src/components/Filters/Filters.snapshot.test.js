import React from 'react';
import Filters from './';
import renderer from 'react-test-renderer';
import { createStore } from 'redux'
import { Provider } from 'react-redux'

const store = createStore(() => ({}));
const stubFunc = () => 'app';

it('initial render', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Filters store={store} onSubmit={stubFunc} />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
