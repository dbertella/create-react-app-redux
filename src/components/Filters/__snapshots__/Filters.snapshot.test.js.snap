exports[`test initial render 1`] = `
<form
  onSubmit={[Function bound submit]}>
  Filters:
  <ul>
    <li>
      age range
      <input
        name="ageMin"
        onBlur={[Function bound submit]}
        onChange={[Function anonymous]}
        onDragStart={[Function anonymous]}
        onDrop={[Function anonymous]}
        onFocus={[Function anonymous]}
        placeholder="min"
        type="text"
        value="" />
      <input
        name="ageMax"
        onBlur={[Function bound submit]}
        onChange={[Function anonymous]}
        onDragStart={[Function anonymous]}
        onDrop={[Function anonymous]}
        onFocus={[Function anonymous]}
        placeholder="min"
        type="text"
        value="" />
    </li>
    <li>
      rate range
      <input
        name="rateMin"
        onBlur={[Function bound submit]}
        onChange={[Function anonymous]}
        onDragStart={[Function anonymous]}
        onDrop={[Function anonymous]}
        onFocus={[Function anonymous]}
        placeholder="min"
        type="text"
        value="" />
      <input
        name="rateMax"
        onBlur={[Function bound submit]}
        onChange={[Function anonymous]}
        onDragStart={[Function anonymous]}
        onDrop={[Function anonymous]}
        onFocus={[Function anonymous]}
        placeholder="min"
        type="text"
        value="" />
    </li>
    <li>
      gender:
      <label>
        <input
          checked={true}
          name="gender"
          onBlur={[Function anonymous]}
          onChange={[Function anonymous]}
          onClick={[Function onChange]}
          onDragStart={[Function anonymous]}
          onDrop={[Function anonymous]}
          onFocus={[Function anonymous]}
          type="radio"
          value="" />
        none
      </label>
      <label>
        <input
          checked={false}
          name="gender"
          onBlur={[Function anonymous]}
          onChange={[Function anonymous]}
          onClick={[Function onChange]}
          onDragStart={[Function anonymous]}
          onDrop={[Function anonymous]}
          onFocus={[Function anonymous]}
          type="radio"
          value="M" />
        M
      </label>
      <label>
        <input
          checked={false}
          name="gender"
          onBlur={[Function anonymous]}
          onChange={[Function anonymous]}
          onClick={[Function onChange]}
          onDragStart={[Function anonymous]}
          onDrop={[Function anonymous]}
          onFocus={[Function anonymous]}
          type="radio"
          value="F" />
        F
      </label>
    </li>
  </ul>
  <button
    onClick={[Function anonymous]}
    type="submit">
    Clear Filters
  </button>
</form>
`;
